src/v/cluster/partition_leaders_table.cc

```diff
@@ -240,6 +241,34 @@ ss::future<> partition_leaders_table::update_with_estimates() {
     }
 }
 
+ss::future<partition_leaders_table::leaders_count_t>
+partition_leaders_table::get_leaders_count(
```

Comment:
Good question. Maybe this is referring to the "estimated" leadership?

Reply:
This method is not safe to use, with `ss::maybe_yield` in the loop it may lead to a situation in which topic table will change after `maybe_yield` returns leading to a crash.

What is the purpose for this method ? If we need to wait for topic partitions we have an API in `controller_api`

Reply:
also we are indeed trying to guesstimate leadership when topic is created which may happen before partitions are even present.

Reply:
Note that `topics` is a copy whose lifecycle is tied to this coroutine, which shoudl be safe, though noted it's definitely something to do a double take at and could use a comment...

I'd like to ensure that the partitions are done with their bootstrapping (not just the placement via controller, but actually started the partitions and Raft groups). I'm looking at `controller_api::wait_for_topic` -- does that do the same thing?

Reply:
yes, exactly i think `wait_for_topic` is what you actually need

Reply:
Thanks for the pointer, done